### ISSUE TITLE
WaitForSingleObject block, 

### ISSUE THREAD

**Issue by lib1256** [zeromq/libzmq#1741]
When in WIN 7, a few programs use ZMQ, will lead to get out
block at signalar.cpp 409, use zmq zeromq4-x-4.0.7 release version:
    //  Set TCP_NODELAY on writer socket.
    rc = setsockopt (*w_, IPPROTO_TCP, TCP_NODELAY,
        (char *)&tcp_nodelay, sizeof (tcp_nodelay));
    wsa_assert (rc != SOCKET_ERROR);

```
//  Enter the critical section.
DWORD dwrc = WaitForSingleObject (sync, INFINITE);
zmq_assert (dwrc == WAIT_OBJECT_0);

//  Bind listening socket to signaler port.
rc = bind (listener, (const struct sockaddr*) &addr, sizeof (addr));
```

**Comment by lib1256** [comment]
https://github.com/zeromq/libzmq/issues/767
i also get.

**Comment by hitstergtd** [comment]
Thanks for reporting this issue.

Could you please:
* build environment (msvc, mingw-w64, etc.) and configuration options used?
* provide a test case for this issue and also #767 if possible; this will go a long way
* any other environmental conditions that lead to this issue?
* attempt to use latest libzmq and see if you can reproduce the issue?

**Comment by sigiesec** [comment]
This was changed by #1543 in connection with #765. The synchronization via a global event is no longer used unless you change config.hpp manually.
libzmq 4.1.5 also contains the fix via zeromq/zeromq4-1#124
